### DISCUSSION TITLE
Can pyright customize code hints?

### DISCUSSION THREAD

**Discussion by wrf1998** [microsoft/pyright#7886] [Q&A]
I am using pyright as the code prompt of the code editor, but there are some packages in my side that are not maintained by python, but are internal packages of our company, so the corresponding tips cannot be displayed when entering the corresponding code. May I ask pyright how to customize these internal packages?
For example: when I type impo, I will be prompted to import, but when I type drag, I will not use dragon, because dragon is our internal package, and I also want to show some examples of the use of this package and some other information when the mouse is hovering over it

I used the lsp server and deployed the service using node.

```
const baseDir = resolve(getLocalDirectory(import.meta.url));
const relativeDir = '../../../../../node_modules/pyright/dist/pyright-langserver.js';
runPythonServer(baseDir, relativeDir);
```

```
import {resolve} from 'path';
import {runLanguageServer} from '../../common/language-server-runner.js';
import {LanguageName} from '../../common/server-commons.js';

export const runPythonServer = (baseDir: string, relativeDir: string) => {
    const processRunPath = resolve(baseDir, relativeDir);
    const port = Number(process.env.port) || Number(process.env.AUTO_PORT0) || 30001;
    console.log('port:' + port);
    runLanguageServer({
        serverName: 'PYRIGHT',
        pathName: '/pyright',
        serverPort: port,
        runCommand: LanguageName.node,
        runCommandArgs: [
            processRunPath,
            '--stdio'
        ],
        wsServerOptions: {
            noServer: true,
            perMessageDeflate: false,
            clientTracking: true
        }
    });
}
```

**Comment by erictraut** [comment]
If I understand you correctly, you are looking for auto-import suggestions. The functionality you're looking for is built into Pylance, which is Microsoft's premium Python language server. It is built on top of pyright. Pylance includes functionality that scans all of the files in your project and in your python environment. It indexes the symbols that can be imported and presents them as auto-import suggestions.

Pyright has some basic language server features, but it is focused on type checking and type analysis. We don't plan to add any new language server features to pyright.

Pylance is available in the VS Code marketplace at no cost.
